src/uuid.cpp

```diff
@@ -0,0 +1,87 @@
+#include "uuid.h"
+
+#include <random>
+#include <limits>
+#include <sstream>
+#include <iomanip>
+
+static std::random_device rd;
+static std::mt19937 eng;
+
+/** use a distribution as mt19937 only guarantees minimum (not exactly) 32 bits */
+static std::uniform_int_distribution<> distrib( 0, std::numeric_limits<uint32_t>::max() );
+
+static inline void gen_uuid4( uint32_t *data ) {
+	data[ 0 ] = distrib( eng );
+	data[ 1 ] = distrib( eng );
+	data[ 2 ] = distrib( eng );
+	data[ 3 ] = distrib( eng );
+}
+
+void uuid::init() {
+	eng.seed( rd() );
+}
+
+uuid::uuid() {
+	gen_uuid4( data );
+}
+
+uuid::uuid( const uuid & ) {
+	gen_uuid4( data );
+}
+
+uuid &uuid::operator=( const uuid & ) {
```

Comment:
See above for the rationale. Moving this to a custom `item` copy constructor wouldn't help. Ideally `item` should be move-only which would solve a lot of our item ownership problems but that's an enormous task and as you commented above could result in some memory fragmentation as we'd need to use pointer indirection.

Ultimately this method is guaranteed to prevent accidental reuse of UUID's. Some documentation could help of course.